### ISSUE TITLE
generate execution trees for lsbs

### ISSUE THREAD

**Issue by pmrowla** [pmrowla/pylivemaker#47]
lmgraph supports generating call trees on a broad per LSB file level, but it would be good to actually have trees for LSB command execution.

`LMScript.walk()` should actually walk a tree (like os.walk), and yield blocks of commands at a time until any conditional or unconditional branch is reached, rather than the current naive search implementation. then the caller can choose which of the branches to actually follow (or none at all).

basically now that we have the text and menu apis, we should be able to generate actual call trees for lsb's, showing the order scripts are actually run, and how branch points work (including both variable/flag based and menu based branching)

**Comment by Stefan311** [comment]
This sounds very interesting. The game I have is very complex. They made a kind of open world adventure game in this visual novel engine...

**Comment by pmrowla** [comment]
@Stefan311 have you tried using lmgraph? you can at least see the lsb scripts call each other already

**Comment by pmrowla** [comment]
`lmgraph` now has two separate sub-commands for generating graphviz DOT syntax graphs

* `lmgraph game` works the same as old `lmgraph`. you pass it `ゲームメイン.lsb` and it generates a DOT call graph for all the user-generated LSBs in a game.
* `lmgraph lsb` generates a DOT execution graph for the commands within an LSB file.

`LMScript.walk()` has not been updated to use the new graph functionality yet

the output DOT files can be converted to viewable formats via graphviz, or in online graphviz viewers

ex `lmgraph lsb` output (after converting an output DOT file to png w/graphviz):

![ex-graph1](https://user-images.githubusercontent.com/651988/82140204-fc970d80-9868-11ea-8543-8f0d00c2eb5f.png)
...
![ex-graph2](https://user-images.githubusercontent.com/651988/82140212-0c165680-9869-11ea-89e0-93ee022d1d4a.png)
